plugin-server/src/main/ingestion-queues/queue.ts

```diff
@@ -28,34 +29,42 @@ export function pauseQueueIfWorkerFull(
 export async function startQueues(
     server: Hub,
     piscina: Piscina,
-    workerMethods: Partial<WorkerMethods> = {}
+    workerMethods: Partial<WorkerMethods> = {},
+    pluginServerMode: PluginServerMode = PluginServerMode.Ingestion
 ): Promise<Queues> {
+    const workerMethodsForMode =
+        pluginServerMode === PluginServerMode.Ingestion
+            ? {
+                  processEvent: (event: PluginEvent) => {
+                      server.lastActivity = new Date().valueOf()
+                      server.lastActivityType = 'processEvent'
+                      return piscina.run({ task: 'processEvent', args: { event } })
+                  },
+                  ingestEvent: (event: PluginEvent) => {
+                      server.lastActivity = new Date().valueOf()
+                      server.lastActivityType = 'ingestEvent'
+                      return piscina.run({ task: 'ingestEvent', args: { event } })
+                  },
+                  onAction: (action: Action, event: PluginEvent) => {
```

Comment:
This should be part of the runner. I realized this might require a bit more of a refactor, so we can move it there later as a follow-up

Reply:
Yes - skipped this for now as it's a larger refactor. Will touch a bunch more files + maybe needs a new topic and consumer.

Not noted anywhere on GH I guess but the plan was always to move this as a step 2.

Also this is not as big of a problem, can stay here for now.